tests/e2e-tests/config/jest.setup.js

```diff
@@ -46,97 +45,48 @@ const OBSERVED_CONSOLE_MESSAGE_TYPES = {
-
-	// If this selector doesn't exist there are no products for us to delete.
-	const bulkSelector = await page.$( '#bulk-action-selector-top' );
-	if ( ! bulkSelector ) {
-		return;
-	}
-
-	// Select all products.
-	await page.waitForSelector( '#cb-select-all-1' );
-	await page.click( '#cb-select-all-1' );
-	// Select the "bulk actions" > "trash" option.
-	await page.select( '#bulk-action-selector-top', 'trash' );
-	// Submit the form to send all draft/scheduled/published posts to the trash.
-	await page.click( '#doaction' );
-	await page.waitForXPath(
-		'//*[contains(@class, "updated notice")]/p[contains(text(), "moved to the Trash.")]'
-	);
-	await switchUserToTest();
-}
-
 /**
  * Navigates to woocommerce import page and imports sample products.
  *
  * @return {Promise} Promise resolving once products have been imported.
  */
 async function importSampleProducts() {
 	await switchUserToAdmin();
-	// Visit Import Products page.
-	await visitAdminPage(
-		'edit.php',
-		'post_type=product&page=product_importer'
-	);
-	await page.click( 'a.woocommerce-importer-toggle-advanced-options' );
-	await page.focus( '#woocommerce-importer-file-url' );
-	// local path for sample data that is included with woo.
-	await page.keyboard.type(
-		'wp-content/plugins/woocommerce/sample-data/sample_products.csv'
-	);
-	await page.click( '.wc-actions .button-next' );
-	await page.waitForSelector( '.wc-importer-mapping-table' );
-	await page.select(
-		'.wc-importer-mapping-table tr:nth-child(29) select',
-		''
+	// Visit `/wp-admin/edit.php?post_type=product` so we can see a list of products and decide if we should import them or not.
+	await visitAdminPage( 'edit.php', 'post_type=product' );
+	const emptyState = await page.evaluate( () =>
+		window.find( 'Ready to start selling something awesome' )
 	);
-	await page.click( '.wc-actions .button-next' );
-	await page.waitForXPath(
-		"//*[@class='woocommerce-importer-done' and contains(., 'Import complete! ')]"
+	const noProduct = await page.evaluate( () =>
+		window.find( 'No products found' )
 	);
-	await switchUserToTest();
+	if ( emptyState || noProduct ) {
+		// Visit Import Products page.
+		await visitAdminPage(
+			'edit.php',
+			'post_type=product&page=product_importer'
+		);
+		await page.click( 'a.woocommerce-importer-toggle-advanced-options' );
+		await page.focus( '#woocommerce-importer-file-url' );
+		// local path for sample data that is included with woo.
+		await page.keyboard.type(
+			'wp-content/plugins/woocommerce/sample-data/sample_products.csv'
+		);
+		await page.click( '.wc-actions .button-next' );
+		await page.waitForSelector( '.wc-importer-mapping-table' );
+		await page.select(
+			'.wc-importer-mapping-table tr:nth-child(29) select',
+			''
+		);
+		await page.click( '.wc-actions .button-next' );
+		await page.waitForXPath(
+			"//*[@class='woocommerce-importer-done' and contains(., 'Import complete! ')]"
+		);
+		await switchUserToTest();
+	}
```

Comment:
Is there not any concern there that we pollute data between test runs if we're not resetting the known state between tests runs in the db?

Reply:
this is a valid concern, but for our case, we're not changing or manipulating products in any way in our tests, we're just querying them, so the initial state of our products should be the same when we finish the tests, and between each test, if a test must change the data shape, that test should be responsible for resetting it.